app/app/components/EditProfile.jsx

```diff
@@ -1,14 +1,17 @@
 import React, { useState } from "react";
+import axios from "axios";
 import styles from "./EditProfile.module.css";
 import Button from "./Button";
 
-const EditProfile = ({ isOpen, onClose }) => {
+const EditProfile = ({ isOpen, onClose, userData, onSave }) => {
     const [formData, setFormData] = useState({
-        profileImageUrl: "",
-        username: "",
-        booksReadCount: "",
-        favoriteBooks: "",
+        firstName: userData?.first_name || "",
+        lastName: userData?.last_name || "",
+        username: userData?.username || "",
+        about: userData?.about || "",
```

Comment:
there is a big change userData is not going to be available on state load. I would suggest listening to userData prop in useEffect and updating then, unless there is something better out there :)